go.work

```diff
@@ -1,7 +1,7 @@
-go 1.18
+go 1.21
```

Comment:
Noting that this is a big jump. I think it's the right time to do it, but it will also be worth calling out to the rest of the community that we require Go v1.21 or newer now.

Reply:
Should be addressed by https://github.com/hyperledger/firefly/pull/1440/commits/ac44d47be7000b1f2ae277040775db8998858696. The changes to make FireFly use 1.21 consistently are more far reaching than just the workflows so have updated those places.

The new images in the commit come from both the golang and alpine image listings:
 - https://hub.docker.com/_/golang 
 - https://hub.docker.com/_/alpine